### PR TITLE
Add instructions for Mac Apple Silicon

### LINKED ISSUE #261: Add instructions for Mac Apple Silicon in the developer section
Rucio does not provide packages and containers for Apple Silicon.

But a Mac with Apple silicon is capable of running code compiled for the x86_64 instruction set using a translation mechanism called Rosetta 2.

In order to have rucio containers working on Apple Silicon you need to:
-  Install rosetta 2
- Set DOCKER_DEFAULT_PLATFORM=linux/amd64

Modifications in the documentation needs to cover this steps. 

Additionally some modifications on VSCode repository: https://github.com/maany/rucio-vscode-dev-env

### PR DESCRIPTION
**Pull request by esilvaju**
Fixes #261

### CODE REVIEW

**Comment by esilvaju** [comment]
On VSCode side I've submitted two PRs to @maany repository which I assume is enough to have the environment ready for that instructions:
https://github.com/maany/rucio-vscode-dev-env/pull/4
https://github.com/maany/rucio-vscode-dev-env/pull/6

On IntelliJ I believe a reference to this section would be enough. Please @rcarpa correct me if I'm wrong.

**Review by rcarpa — APPROVED** [review]

File: docs/developer/setting_up_mac_apple_silicon.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+In order to force docker to run commands with platform **linux/amd64** instead of **linux/arm64** by default on macOS Apple Silicon, you have two options:
+
+Set Docker default plataform to linux/amd64:
+
+    export DOCKER_DEFAULT_PLATFORM=linux/amd64
+
+Or run it as part of the command a single time:
+
+    DOCKER_DEFAULT_PLATFORM=linux/amd64 docker-compose -f <docker-compose-file.yml>
```
Comment (rcarpa) [inline docs/developer/setting_up_mac_apple_silicon.md:30]: I wonder if we should just add the platform to the docker-compose file

File: website/sidebars.json [hunk 2]
```diff
@@ -28,6 +28,7 @@
                 "release-notes"
             ],
             "User": [
+                "user/setting_up_mac_apple_silicon",
```
Comment (rcarpa) [inline website/sidebars.json:31]: shouldn't it be only for the developer ? you didn't add this particular file

Comment (bari12) [reply website/sidebars.json:31]: Yes, this needs to be removed. 😄

File: docs/developer/setting_up_mac_apple_silicon.md (hunk 1 above)
Comment (esilvaju) [reply docs/developer/setting_up_mac_apple_silicon.md:30]: I did this test and only worked setting like this.

File: website/sidebars.json (hunk 2 above)
Comment (esilvaju) [reply website/sidebars.json:31]: Removed ;)

Comment (bari12) [reply website/sidebars.json:31]: Does not seem to be removed :-D

**Pull request closed without merging**